Userland/Libraries/LibWeb/CMakeLists.txt

```diff
@@ -242,70 +242,51 @@ set(GENERATED_SOURCES
-        MAIN_DEPENDENCY ${class}.idl
-    )
-    add_custom_command(
-        OUTPUT Bindings/${basename}Constructor.cpp
-        COMMAND ${write_if_different} Bindings/${basename}Constructor.cpp CodeGenerators/WrapperGenerator --constructor-implementation ${CMAKE_CURRENT_SOURCE_DIR}/${class}.idl
-        VERBATIM
-        DEPENDS WrapperGenerator
-        MAIN_DEPENDENCY ${class}.idl
-    )
-    add_custom_command(
-        OUTPUT Bindings/${basename}Prototype.h
-        COMMAND ${write_if_different} Bindings/${basename}Prototype.h CodeGenerators/WrapperGenerator --prototype-header ${CMAKE_CURRENT_SOURCE_DIR}/${class}.idl
-        VERBATIM
-        DEPENDS WrapperGenerator
-        MAIN_DEPENDENCY ${class}.idl
+    get_filename_component(basename "${class}" NAME)
+    set(BINDINGS_SOURCES
+        "Bindings/${basename}Wrapper.h"
+        "Bindings/${basename}Wrapper.cpp"
+        "Bindings/${basename}Constructor.h"
+        "Bindings/${basename}Constructor.cpp"
+        "Bindings/${basename}Prototype.h"
+        "Bindings/${basename}Prototype.cpp"
     )
-    add_custom_command(
-        OUTPUT Bindings/${basename}Prototype.cpp
-        COMMAND ${write_if_different} Bindings/${basename}Prototype.cpp CodeGenerators/WrapperGenerator --prototype-implementation ${CMAKE_CURRENT_SOURCE_DIR}/${class}.idl
-        VERBATIM
-        DEPENDS WrapperGenerator
-        MAIN_DEPENDENCY ${class}.idl
+    set(BINDINGS_TYPES
+        header
+        implementation
+        constructor-header
+        constructor-implementation
+        prototype-header
+        prototype-implementation
     )
-    add_custom_target(generate_${basename}Wrapper.h DEPENDS Bindings/${class}Wrapper.h)
-    add_custom_target(generate_${basename}Wrapper.cpp DEPENDS Bindings/${class}Wrapper.cpp)
-    add_custom_target(generate_${basename}Constructor.h DEPENDS Bindings/${class}Constructor.h)
-    add_custom_target(generate_${basename}Constructor.cpp DEPENDS Bindings/${class}Constructor.cpp)
-    add_custom_target(generate_${basename}Prototype.h DEPENDS Bindings/${class}Prototype.h)
-    add_custom_target(generate_${basename}Prototype.cpp DEPENDS Bindings/${class}Prototype.cpp)
+    target_sources(LibWeb PRIVATE ${BINDINGS_SOURCES})
+    # FIXME: cmake_minimum_required(3.17) for ZIP_LISTS
+    list(LENGTH BINDINGS_SOURCES num_bindings)
+    math(EXPR bindings_end "${num_bindings} - 1")
+    foreach(iter RANGE "${bindings_end}")
+        list(GET BINDINGS_SOURCES ${iter} bindings_src)
+        list(GET BINDINGS_TYPES ${iter} bindings_type)
+        add_custom_command(
+            OUTPUT "${bindings_src}"
+            BYPRODUCTS "${bindings_src}.tmp"
+            COMMAND "$<TARGET_FILE:WrapperGenerator>" "--${bindings_type}" "${CMAKE_CURRENT_SOURCE_DIR}/${class}.idl" > "${bindings_src}.tmp"
+            COMMAND "${CMAKE_COMMAND}" -E copy_if_different "${bindings_src}.tmp" "${bindings_src}"
+            COMMAND "${CMAKE_COMMAND}" -E remove "${bindings_src}.tmp"
```

Comment:
Ok now I'm confused. What is it about the change to using the copy_if_different and remove that causes these commands to run every single time you invoke `ninja` or `ninja install` ? Kind of a bummer to create 753 files  just to do no work in the end...

Reply:
You still have the `BYPRODUCTS` there. The build system sees that the byproducts are missing and so regenerates them.

Reply:
Oh! That makes a lot of sense then. I've removed that now 🤦‍♂️ .